protocol/kona-node-arch.md

```diff
@@ -0,0 +1,291 @@
+
+## Architecture
+
+### Building Blocks from First Principles
+
+The Rollup Node is a core component of the OP Stack.
+It is responsible for constructing the canonical safe L2 chain.
+In order to produce the chain, the Rollup Node listens to two sources
+of information.
+
+1. **Data Availability Layer**: As new L1 blocks are produced, the receipts
+are parsed for new L2 chain updates. L2 inputs (L2 transaction batches + deposits)
+are then derived from the data availability (DA) layer.
+
+2. **L2 Sequencer**: The L2 sequencer produces unsafe L2 blocks and sends them over
+p2p gossip to other rollup nodes.
+
+What this looks like currently is two sources feeding into derivation,
+somehow producing the L2 Chain.
+
+```
+┌────────────┐
+│L2 Sequencer│
+│            ├────────────────────┐
+│   Gossip   │                    │    ┌────────────┐
+└────────────┘                    │    │            │
+                                  ├──► │     ???    │──►  < L2 Chain >
+┌────────────┐    ┌────────────┐  │    │            │
+│     DA     │    │            │  │    └────────────┘
+│            ├──► │ Derivation ├──┘
+│   Watcher  │    │            │
+└────────────┘    └────────────┘
+```
+
+From these sources, the rollup node imports "unsafe" blocks from the L2 sequencer
+and safe blocks from the L2 derivation pipeline. Both unsafe and safe blocks
+are imported into the L2 execution layer via the Engine API.
+
+```
+┌────────────┐
+│L2 Sequencer│
+│            ├────────────────────┐
+│   Gossip   │                    │    ┌────────────┐
+└────────────┘                    │    │            │
+                                  ├──► │ Engine API │──►  < L2 Chain >
+┌────────────┐    ┌────────────┐  │    │            │
+│     DA     │    │            │  │    └────────────┘
+│            ├──► │ Derivation ├──┘
+│   Watcher  │    │            │
+└────────────┘    └────────────┘
+```
+
+The [Holocene Hardfork][holocene] introduced steady block derivation.
+This change allows a payload to be replaced with a deposits-only block
+to allow the L2 chain to progress even if an invalid payload is derived.
+
+Steady block derivation has significant architectural impacts for the
+rollup node. With Holocene, the derivation component needs to be updated
+if a payload is replaced with a deposits-only block. This requires the
+Engine API component to send a message to the derivation component,
+introducing an edge in the modular component graph.
+
+[holocene]: https://specs.optimism.io/protocol/holocene/derivation.html
+
+```
+┌────────────┐
+│L2 Sequencer│
+│            ├───┐
+│   Gossip   │   │   ┌────────────┐   ┌────────────┐
+└────────────┘   │   │            │   │            │
+                 ├──►│ Derivation │──►│ Engine API │──►  < L2 Chain >
+┌────────────┐   │   │            │   │            │
+│     DA     │   │   └────────────┘   └──────┬─────┘
+│            ├───┘          ▲                │
+│   Watcher  │              └────────────────┘
+└────────────┘
```

Comment:
In previous illustrations, the Unsafe Gossip was being routed directly to the Engine API. I *think* that is still the case here, even when Holocene introduces block replacement, ya?

And actually, I am not sure: do block replacements occur on invalid *unsafe* blocks? I would assume not since they are not part of derivation (and therefore not part of *steady* derivation).

Reply:
Oops good catch - as discussed on the call, will update this. Bad diagram on my end 😅

Reply:
Updated!